src/components/text/index.test.tsx

```diff
@@ -394,4 +394,34 @@ describe("<Text /> component functional testing", () => {
     // eslint-disable-next-line no-console
     (console.error as jest.Mock).mockRestore();
   });
+  it("Should update the dataset in context after saving", async () => {
+    jest.spyOn(SolidFns, "saveSolidDatasetAt").mockResolvedValue(savedDataset);
+    const setDataset = jest.fn();
+    const setThing = jest.fn();
+    jest.spyOn(helpers, "useProperty").mockReturnValue({
+      dataset: mockDataset,
+      setDataset,
+      setThing,
+      error: undefined,
+      value: mockNick,
+      thing: mockThing,
+      property: mockPredicate,
+    });
+    const { getByDisplayValue } = render(
+      <Text
+        solidDataset={mockDatasetWithResourceInfo}
+        thing={mockThing}
+        property={mockPredicate}
+        locale="en"
+        edit
+        autosave
+      />
+    );
+    const input = getByDisplayValue(mockNick);
+    input.focus();
+    fireEvent.change(input, { target: { value: "updated nick value" } });
+    input.blur();
+    expect(SolidFns.saveSolidDatasetAt).toHaveBeenCalled();
```

Comment:
I believe the issue here is that by calling `saveSolidDatasetAt` and passing the returned dataset to the state using `setState` is that the SolidDataset returned by this function (saveSolidDatasetAt) will contain the data **sent to the Pod** and what we want to do instead is to call `getSolidDataset` again after successfully saving the data. Then we set the newly fetched one to the state to avoid any potential issues of not having the latest data

[Module: resource/solidDataset — Inrupt solid-client API Documentation ](https://docs.inrupt.com/developer-tools/api/javascript/solid-client/modules/resource_soliddataset.html#savesoliddatasetat)

  I believe this process is documented in the description on the above link.

Reply:
This is correct, we'll need to run a `get` after save. Not ideal, but it's the safest option.